[moxy] jaxb-compiler command fails to find jaxb implementation

running `./jaxb-compiler.sh ../xsds/beanval.xsd ` fails with:

```
Exception in thread "main" java.lang.AssertionError: jakarta.xml.bind.JAXBException: Implementation of Jakarta XML Binding-API has not been found on module path or classpath.
 - with linked exception:
[java.lang.ClassNotFoundException: org.glassfish.jaxb.runtime.v2.ContextFactory]
	at com.sun.tools.xjc.reader.xmlschema.bindinfo.BindInfo.getCustomizationContext(BindInfo.java:326)
...
Caused by: jakarta.xml.bind.JAXBException: Implementation of Jakarta XML Binding-API has not been found on module path or classpath.
 - with linked exception:
[java.lang.ClassNotFoundException: org.glassfish.jaxb.runtime.v2.ContextFactory]
	at jakarta.xml.bind.ContextFinder.newInstance(ContextFinder.java:255)
	at jakarta.xml.bind.ContextFinder.newInstance(ContextFinder.java:243)
	at jakarta.xml.bind.ContextFinder.find(ContextFinder.java:398)
	at jakarta.xml.bind.JAXBContext.newInstance(JAXBContext.java:691)
	at jakarta.xml.bind.JAXBContext.newInstance(JAXBContext.java:632)
	at com.sun.tools.xjc.reader.xmlschema.bindinfo.BindInfo.getCustomizationContext(BindInfo.java:306)
	... 62 more
Caused by: java.lang.ClassNotFoundException: org.glassfish.jaxb.runtime.v2.ContextFactory
...
```

the problem here is that XJC contains binding classes representing customization schema. When it tries to initialize them through the `JAXBContext.newInstance(Class...)` API call, no xml binding implementation is found because XJC as such does not depend on the specific runtime - it should just find MOXy and use that.

Possibilities:
* add definition to `META-INF/services` - if it is added, then it is likely that this entry would hide or compete with the default/other implementation and we don't want that. We want to allow both implementations to co-exist ie in the server environment
* add `jaxb.properties` to `com.sun.tools.xjc.reader.xmlschema.bindinfo` package - a side effect of this can be that even default/other implementation could read the model through MOXy - again, something we probably don't want
* pass `jakarta.xml.bind.JAXBContextFactory` system option to XJC through the script - probably the best option; the command itself is there for the use from command line, so it is unlikely it will interfere with others when run

--- a/bundles/eclipselink/src/main/scripts/bin/jaxb-compiler.cmd
+++ b/bundles/eclipselink/src/main/scripts/bin/jaxb-compiler.cmd
@@ -1,5 +1,5 @@
 @REM
-@REM Copyright (c) 2018, 2020 Oracle and/or its affiliates. All rights reserved.
+@REM Copyright (c) 2018, 2021 Oracle and/or its affiliates. All rights reserved.
 @REM
 @REM This program and the accompanying materials are made available under the
 @REM terms of the Eclipse Public License v. 2.0 which is available at
@@ -15,7 +15,7 @@
 call "%~dp0setenv.cmd"
 
 @REM User may increase Java memory setting(s) if desired:
-set JVM_ARGS=-Xmx256m
+set JVM_ARGS=-Xmx256m -Djakarta.xml.bind.JAXBContextFactory=org.eclipse.persistence.jaxb.JAXBContextFactory
 
 @REM If going through a proxy, set the proxy host and proxy port below, then uncomment the line
 @REM set JVM_ARGS=%JVM_ARGS% -DproxySet=true -Dhttp.proxyHost= -Dhttp.proxyPort=

--- a/bundles/eclipselink/src/main/scripts/bin/jaxb-compiler.sh
+++ b/bundles/eclipselink/src/main/scripts/bin/jaxb-compiler.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 #
-# Copyright (c) 2018, 2020 Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2018, 2021 Oracle and/or its affiliates. All rights reserved.
 #
 # This program and the accompanying materials are made available under the
 # terms of the Eclipse Public License v. 2.0 which is available at
@@ -14,7 +14,7 @@
 . `dirname $0`/setenv.sh
 
 # User may increase Java memory setting(s) if desired:
-JVM_ARGS="-Xmx256m"
+JVM_ARGS="-Xmx256m -Djakarta.xml.bind.JAXBContextFactory=org.eclipse.persistence.jaxb.JAXBContextFactory"
 
 # If going through a proxy, set the proxy host and proxy port below, then uncomment the line
 # JVM_ARGS="${JVM_ARGS} -DproxySet=true -Dhttp.proxyHost= -Dhttp.proxyPort="

--- a/bundles/moxy-standalone/src/main/scripts/jaxb-compiler.cmd
+++ b/bundles/moxy-standalone/src/main/scripts/jaxb-compiler.cmd
@@ -1,5 +1,5 @@
 @REM
-@REM Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+@REM Copyright (c) 2020, 2021 Oracle and/or its affiliates. All rights reserved.
 @REM
 @REM This program and the accompanying materials are made available under the
 @REM terms of the Eclipse Public License v. 2.0 which is available at
@@ -15,7 +15,7 @@
 call "%~dp0setenv.cmd"
 
 @REM User may increase Java memory setting(s) if desired:
-set JVM_ARGS=-Xmx256m
+set JVM_ARGS=-Xmx256m -Djakarta.xml.bind.JAXBContextFactory=org.eclipse.persistence.jaxb.JAXBContextFactory
 
 @REM If going through a proxy, set the proxy host and proxy port below, then uncomment the line
 @REM set JVM_ARGS=%JVM_ARGS% -DproxySet=true -Dhttp.proxyHost= -Dhttp.proxyPort=

--- a/bundles/moxy-standalone/src/main/scripts/jaxb-compiler.sh
+++ b/bundles/moxy-standalone/src/main/scripts/jaxb-compiler.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 #
-# Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2020, 2021 Oracle and/or its affiliates. All rights reserved.
 #
 # This program and the accompanying materials are made available under the
 # terms of the Eclipse Public License v. 2.0 which is available at
@@ -14,7 +14,7 @@
 . `dirname $0`/setenv.sh
 
 # User may increase Java memory setting(s) if desired:
-JVM_ARGS="-Xmx256m"
+JVM_ARGS="-Xmx256m -Djakarta.xml.bind.JAXBContextFactory=org.eclipse.persistence.jaxb.JAXBContextFactory"
 
 # If going through a proxy, set the proxy host and proxy port below, then uncomment the line
 # JVM_ARGS="${JVM_ARGS} -DproxySet=true -Dhttp.proxyHost= -Dhttp.proxyPort="

--- a/bundles/moxy-standalone/src/main/scripts/setenv.cmd
+++ b/bundles/moxy-standalone/src/main/scripts/setenv.cmd
@@ -1,0 +1,37 @@
+@REM
+@REM Copyright (c) 2018, 2021 Oracle and/or its affiliates. All rights reserved.
+@REM
+@REM This program and the accompanying materials are made available under the
+@REM terms of the Eclipse Public License v. 2.0 which is available at
+@REM http://www.eclipse.org/legal/epl-2.0,
+@REM or the Eclipse Distribution License v. 1.0 which is available at
+@REM http://www.eclipse.org/org/documents/edl-v10.php.
+@REM
+@REM SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+@REM
+
+@REM User MUST set JAVA_HOME to point a supported JRE. If none
+@REM is provided for INSTALL_JAVA_HOME then the system JAVA_HOME
+@REM value will be used
+set INSTALL_JAVA_HOME=%s_jreDirectory%
+if "%JAVA_HOME%"=="" (
+    set JAVA_HOME=%INSTALL_JAVA_HOME%
+)
+
+@REM Please do not change any of the following lines:
+
+set _FIXPATH=
+call :fixpath "%JAVA_HOME%"
+set JAVA_HOME=%_FIXPATH:~1%
+goto :EOF
+
+:fixpath
+if not %1.==. (
+  for /f "tokens=1* delims=;" %%a in (%1) do (
+    call :shortfilename "%%a" & call :fixpath "%%b"
+  )
+)
+goto :EOF
+:shortfilename
+for %%i in (%1) do set _FIXPATH=%_FIXPATH%;%%~fsi
+goto :EOF

--- a/bundles/moxy-standalone/src/main/scripts/setenv.sh
+++ b/bundles/moxy-standalone/src/main/scripts/setenv.sh
@@ -1,0 +1,20 @@
+#!/bin/sh
+#
+# Copyright (c) 2018, 2021 Oracle and/or its affiliates. All rights reserved.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License v. 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0,
+# or the Eclipse Distribution License v. 1.0 which is available at
+# http://www.eclipse.org/org/documents/edl-v10.php.
+#
+# SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+#
+
+# User MUST set INSTALL_JAVA_HOME to point a supported JRE. If none
+# is provided for INSTALL_JAVA_HOME then the system JAVA_HOME
+# value will be used
+INSTALL_JAVA_HOME=%s_jreDirectory%;
+if  [ "${JAVA_HOME}" = '' ]; then
+    JAVA_HOME=$INSTALL_JAVA_HOME; export JAVA_HOME
+fi

